src/analysis/convex/basic.lean

```diff
@@ -881,18 +879,85 @@ lemma concave_on.le_on_segment {γ : Type*}
     min (f x) (f y) ≤ f z :=
 @convex_on.le_on_segment _ _ _ _ (order_dual γ) _ _ _ f hf x y z hx hy hz
 
+-- could be shown without contradiction but yeah
+lemma convex_on.le_left_of_right_le' (hf : convex_on s f) {x y : E} {a b : ℝ}
+  (hx : x ∈ s) (hy : y ∈ s) (ha : 0 < a) (hb : 0 ≤ b) (hab : a + b = 1)
+  (hxy : f y ≤ f (a • x + b • y)) :
+  f (a • x + b • y) ≤ f x :=
+begin
+  apply le_of_not_lt (λ h, lt_irrefl (f (a • x + b • y)) _),
+  calc
+    f (a • x + b • y)
+        ≤ a • f x + b • f y : hf.2 hx hy ha.le hb hab
+    ... < a • f (a • x + b • y) + b • f (a • x + b • y)
+        : add_lt_add_of_lt_of_le (smul_lt_smul_of_pos h ha) (smul_le_smul_of_nonneg hxy hb)
+    ... = f (a • x + b • y) : by rw [←add_smul, hab, one_smul],
+end
+
+lemma concave_on.left_le_of_le_right' (hf : concave_on s f) {x y : E} {a b : ℝ}
+  (hx : x ∈ s) (hy : y ∈ s) (ha : 0 < a) (hb : 0 ≤ b) (hab : a + b = 1)
+  (hxy : f (a • x + b • y) ≤ f y) :
+  f x ≤ f (a • x + b • y) :=
+@convex_on.le_left_of_right_le' _ _ _ _ (order_dual γ) _ _ _ f hf x y a b hx hy ha hb hab hxy
+
+lemma convex_on.le_right_of_left_le' (hf : convex_on s f) {x y : E} {a b : ℝ}
```

Comment:
Can you prove this with `convex_on.le_left_of_right_le'` by swapping the role of `x/y` and `a/b`?

Reply:
Oh yeah actually it's just `rw add_comm at ⊢ hab hxy`. Was expecting it to be harder.